src/domain/relay/relay.repository.ts

```diff
@@ -56,7 +51,28 @@ export class RelayRepository {
       }
     }
 
-    return this.relayApi.relay(relayPayload);
+    const relayResponse = await this.relayApi.relay(relayPayload);
+
+    // If we fail to increment count, we should not fail the relay
+    await Promise.allSettled(
+      relayAddresses.map((address) => {
+        return this.incrementRelayCount({
+          chainId: relayPayload.chainId,
+          address,
+        });
+      }),
+    );
```

Comment:
This `Promise.allSettled` will call multiple times `incrementRelayCount` in parallel, right?  

As `incrementRelayCount` increments a counter, theoretically this could lead to a race condition. Maybe I'm missing something, but I'd change this implementation to:

```javascript
    for (const address of relayAddresses) {
      await this.incrementRelayCount({
        chainId: relayPayload.chainId,
        address,
      });
    }
```

Reply:
Good point. I've implemented your suggestion in dc73902b4838bcb57f7a97a74f997cc572677a30.